console/console-init/ui/src/modules/address-space/dialogs/CreateMessagingProject/MessagingProjectConfiguration.tsx

```diff
@@ -0,0 +1,241 @@
+        project.isNameValid = dnsSubDomainRfc1123NameRegexp.test(value);
+        break;
+      case "namespace":
+        project.namespace = value.trim();
+        break;
+      case "isNameValid":
+        project.isNameValid = value === "true" ? true : false;
+        break;
+      case "type":
+        project.type = value.trim();
+        project.plan = "";
+        project.authService = "";
+        break;
+      case "plan":
+        project.plan = value.trim();
+        break;
+      case "authService":
+        project.authService = value.trim();
+        break;
+      case "customizeEndpoint":
+        project.customizeEndpoint = value === "true" ? true : false;
+        setStepStatus({
+          ...stepStatus,
+          showCustomize: project.customizeEndpoint
+        });
+        break;
+    }
+    setProjectDetail(project);
+  };
+
+  const onNameSpaceSelect = (value: string) => {
+    setField("namespace", value);
+  };
+  const handleNameChange = (value: string) => {
+    setField("name", value);
+  };
+  const handleBrokeredChange = () => {
+    // if (projectDetail.type !== "brokered") {
+    setField("type", "brokered");
+    // }
+  };
+  const handleStandardChange = () => {
+    // if (projectDetail.type !== "standard") {
+    setField("type", "standard");
+    // }
+  };
+  const onPlanSelect = (value: string) => {
+    setField("plan", value);
+  };
+  const onAuthenticationServiceSelect = (value: string) => {
+    setField("authService", value);
+  };
+
+  const getNameSpaceOptions = () => {
+    let nameSpaceOptions: any[];
+    nameSpaceOptions = namespaces.map((namespace: any) => {
+      return {
+        value: namespace.metadata.name,
+        label: namespace.metadata.name
+      };
+    });
+    return nameSpaceOptions;
+  };
+  const hanldeCustomEndpointChange = (value: boolean) => {
+    setField("customizeEndpoint", value.toString());
+  };
+
+  const getPlanOptions = () => {
+    let planOptions: any[] = [];
+    if (type) {
+      planOptions =
+        addressSpacePlans
+          .map(plan => {
+            if (plan.spec.addressSpaceType === type) {
+              return {
+                value: plan.metadata.name,
+                label: plan.spec.displayName || plan.metadata.name,
+                description:
+                  plan.spec.shortDescription || plan.spec.longDescription
+              };
+            }
+          })
+          .filter(plan => plan !== undefined) || [];
+    }
+    return planOptions;
+  };
+
+  const getAuthenticationServiceOptions = () => {
+    let authenticationServiceOptions: IAuthenticationServiceOptions[] = [];
+    if (authenticationServices) {
+      authenticationServices.addressSpaceSchema_v2.forEach(
+        (authService: any) => {
+          if (authService.metadata.name === type) {
+            authenticationServiceOptions = authService.spec.authenticationServices.map(
+              (service: any) => {
+                return {
```

Comment:
I think, we don't need extra return statement here, arrow function return itself.